Sources_v3/Database/DatabaseSession.swift

```diff
@@ -79,6 +79,22 @@ extension MessageDatabaseSession {
     }
 }
 
+protocol ChannelDatabaseSession {
+    /// Creates a new `ChannelDTO` object in the database with the given `payload` and `query`.
+    @discardableResult
+    func saveChannel<ExtraData: ExtraDataTypes>(payload: ChannelPayload<ExtraData>, query: ChannelListQuery?) throws -> ChannelDTO
+    
+    /// Creates a new `ChannelDTO` object in the database with the given `payload` and `query`.
+    @discardableResult
+    func saveChannel<ExtraData: ExtraDataTypes>(
+        payload: ChannelDetailPayload<ExtraData>,
+        query: ChannelListQuery?
+    ) throws -> ChannelDTO
+    
+    /// Fetches `ChannelDTO` with the given `cid` from the database.
+    func channel(cid: ChannelId) -> ChannelDTO?
+}
+
```

Comment:
This is actually CIS-332 🤷‍♂️  You can probably close that one with this PR, too :)

Reply:
@evsaev Please add [CIS-332] to the PR description, too.